react/week2/my-app/src/App.js

```diff
@@ -0,0 +1,101 @@
+import React, {useState, useEffect} from "react"
+import TodoItem from "./TodoItem"
+import todosData from "./todosData"
+
+
+class App extends React.Component {
+    constructor(props) {
+        super(props)
+        this.state = {
+            todos: todosData,
+            setTodos: todosData
+        }
+
+        // const [todos, setTodos] = useState(todosDatas);
+        // this.setTodos = setTodos;
+        // this.todos = todos;
+
+        this.handleChange = this.handleChange.bind(this)
+        this.addTodo = this.addTodo.bind(this)
+        this.deleteItem = this.deleteItem.bind(this)
+    }
+
+    handleChange(id) {
+        this.setState(prevState => {
+            const updatedTodos = prevState.todos.map(todo => {
+                if (todo.id === id) {
+                    todo.completed = !todo.completed
+                }
+                return todo
+            })
+            return {
+                todos: updatedTodos
+            }
+        })
+    }
```

Comment:
Does this function works on your machine? It doesn't work when I cloned the homework.
If not try maybe this one:
```suggestion
    this.setState((prevState) => {
      const updatedTodos = prevState.todos.map((todo) => {
        if (todo.id === id) {
          return { ...todo, completed: !todo.completed };
        } else {
          return todo;
        }
      });
      return { todos: updatedTodos };
    });
```